### PR TITLE
Refactor state channels

### PR DESCRIPTION
**Pull request by davidyuk**

### CODE REVIEW

**Comment by davidyuk** [comment]
@mradkov @thepiwo I think this one is ready for review

File: es/channel/internal.js [hunk 1]
```diff
@@ -27,31 +26,24 @@ const PING_TIMEOUT_MS = 10000
...
 const pongTimeoutId = new WeakMap()
-const fsmId = new WeakMap()
+export const fsmId = new WeakMap()
 
-function channelURL (url, params) {
-  const paramString = R.join('&', R.values(R.mapObjIndexed((value, key) =>
-    `${pascalToSnake(key)}=${encodeURIComponent(value)}`, params)))
-
-  return `${url}?${paramString}`
-}
```
Comment (mradkov) [inline es/channel/internal.js:52]: Isn't this used anywhere?

Comment (davidyuk) [reply es/channel/internal.js:52]: It is used only in this file, in `initialize` function, otherwise, it will be rejected by linter.

**Comment by davidyuk** [comment]
> Do we have any example where we can try this?

I have tried this branch in the cont-toss-game (https://github.com/aeternity/coin-toss-game/tree/superhero-demo, https://github.com/aeternity/ae-channel-service/pull/113), it works fine with these changes. 

> Does the test coverage also cover this?

Yes, it is heavily covered in this file: https://github.com/aeternity/aepp-sdk-js/blob/9494365840f47dbad40c515f1e7b6b4f91928ce1/test/integration/channel.js

**Pull request merged**